### ISSUE TITLE
"Incompatible types in assignment" when assigning with a type alias

### ISSUE THREAD

**Issue by FrankBattaglia** [python/mypy#14165]
**Bug Report**

`mypy` appears to give a false positive typing error when assigning a narrowed type alias to a subclass's class attribute (overriding the base class's value).

**To Reproduce**

```python
from collections.abc import Mapping

StringMapping = Mapping[str, str]


class MyBase(object):
    MAPPING_TYPE = Mapping


class MyDirectSubclass(MyBase):
    MAPPING_TYPE = Mapping[str, str]  # no error


class MyIndirectSubclass(MyBase):
    MAPPING_TYPE = StringMapping  # mypy error
```
https://mypy-play.net/?mypy=latest&python=3.11&gist=5d9d71400d5546a125de9de1d452d199

**Expected Behavior**

I expect the type alias to work as the directly specified type, i.e., no errors in either case.  Admittedly I don't always completely grok LSP so maybe I'm missing something subtle...

**Actual Behavior**

> main.py:15: error: Incompatible types in assignment (expression has type "Type[Mapping[Any, Any]]", base class "MyBase" defined the type as "Type[Mapping[Any, Any]]")  [assignment]
Found 1 error in 1 file (checked 1 source file)

**Your Environment**

- Mypy version used: 0.961 (local) / 0.991 (mypy-play.net)
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
```
[mypy]
disallow_untyped_defs = True
show_error_codes = True
``` 
(local)
- Python version used: 3.9.7 (local) / 3.11 (mypy-play.net)

**Comment by erictraut** [comment]
If you change the type alias in `MyBase` to define the type alias as `Mapping[Any, Any]`, the error goes away. 

What appears to be happening here is that mypy is leaving the `Mapping` type unspecialized in this case, so its type is actually `Mapping[_KT, _VT_co]` rather than `Mapping[Any, Any]`. This kind of makes sense because it's not clear whether `MAPPING_TYPE` is a class variable or a type alias (it's ambiguous). If your intent is to declare it as a type alias, you can add a `TypeAlias` annotation, and that also eliminates the error from mypy.

```python
class MyBase(object):
    MAPPING_TYPE: TypeAlias = Mapping
```
